src/users/entities/user.entity.ts

```diff
@@ -0,0 +1,22 @@
+import { Student, Users } from '@prisma/client'
+
+export class UserEntity implements Users {
+    id!: number
+    username!: string
+    profileUrl!: string | null
+    firstnameTh!: string | null
+    lastnameTh!: string | null
+    firstnameEn!: string | null
+    lastnameEn!: string | null
+    nickname!: string | null
+    email!: string
+    phone!: string
+    password!: string
+    roleId!: number
+    createdDate!: Date
+    createdBy!: number
+    updatedDate!: Date
+    updatedBy!: number | null
```

Comment:
split to BaseDocument interface  and implement to this class

Reply:
Can you explain more?

Reply:
https://stackoverflow.com/a/46075261 this way